rpc/observers.go

```diff
@@ -28,6 +28,14 @@ type Observer interface {
 	ServerReply(req Request, hdr *Header, body interface{})
 }
 
+// ObserverFactory is a type which can construct a new Observer.
+type ObserverFactory interface {
+	// RPCObserver will return a new Observer usually constructed
+	// from the state previously built up in the Observer. The
+	// returned instance will be utilized per RPC request.
+	RPCObserver() Observer
```

Comment:
maybe just Observer() ?
there's no need to bring RPC into this interface method, we're already in the rpc package

Reply:
This is an existing interface - I've just moved it here. I think it's RPCObserver to make it clearer when it's embedded in the `apiserver/observer.Observer` interface (rather than it just being Observer.Observer). Agreed that it's a bit awkward though. I can change it, but it'll hit the other observer implementations.